File: leetcode.com/0112_path_sum/solution.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */

bool hasPathSum(struct TreeNode* root, int targetSum)
{
    if (root == NULL)
        return false;

    if (root->left == NULL && root->right == NULL) {
        targetSum -= root->val;

        if (targetSum == 0)
            return true;

        return false;
    }

    if (hasPathSum(root->left, targetSum - root->val))
        return true;

    if (hasPathSum(root->right, targetSum - root->val))
        return true;

    return false;
}

File: leetcode.com/0112_path_sum/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def hasPathSum(self, root: TreeNode, targetSum: int) -> bool:
        if not root:
            return False

        if root.left == root.right == None:
            return (targetSum - root.val == 0)

        if self.hasPathSum(root.left, targetSum - root.val):
            return True

        if self.hasPathSum(root.right, targetSum - root.val):
            return True

        return False

